src/compiler/emitter.ts

```diff
@@ -4716,6 +4719,9 @@ if (typeof __param !== "function") __param = function (paramIndex, decorator) {
                     write(unaliasedModuleNames.join(", "));
                 }
                 write("], function (require, exports");
+                if (compilerOptions.noEmitHelpers) {

```

Comment:
@whitneyit If you look at #2911, what we are considering is making a separate runtime library available that contains these helpers, which would allow you to write the following:

```
tsc --noEmitHelpers --import tslib --module commonjs app.ts
```

So your app.js output might be the following for CommonJS:
```js
require("tslib");
// app body
```

Or the following for AMD:
```js
define(["require", "exports", "tslib"], function (require, exports) {
  // app body
});
```
And the `tslib` library (or whatever it is called), would define globals for `__extends`, `__decorate`, etc.